test/python/compiler/test_transpiler.py

```diff
@@ -1845,6 +1927,29 @@ def test_qasm3_output_control_flow(self, optimization_level):
             str,
         )
 
+    @data(0, 1, 2, 3)
+    def test_qasm3_output_control_flow_expr(self, optimization_level):
+        """Test that the output of a transpiled circuit with control flow and `Expr` nodes can be
+        dumped into OpenQASM 3."""
+        backend = FakeMumbaiV2()
+        backend.target.add_instruction(IfElseOp, name="if_else")
+        backend.target.add_instruction(ForLoopOp, name="for_loop")
+        backend.target.add_instruction(WhileLoopOp, name="while_loop")
+        backend.target.add_instruction(SwitchCaseOp, name="switch_case")
+        transpiled = transpile(
+            self._control_flow_circuit(),
+            backend=backend,
+            optimization_level=optimization_level,
+            seed_transpiler=2023_07_26,
+        )
+        # TODO: There's not a huge amount we can sensibly test for the output here until we can
+        # round-trip the OpenQASM 3 back into a Terra circuit.  Mostly we're concerned that the dump
+        # itself doesn't throw an error, though.
```

Comment:
Heh, well this preempts my comment here. It will be good to update this to test the full roundtrip after the importer has support.

Reply:
Yeah, the test is nearly just copy/pasted from a very similar one above.